libs/server/Storage/Session/ObjectStore/SetOps.cs

```diff
@@ -398,19 +391,57 @@ internal unsafe GarnetStatus SetMove(ArgSlice sourceKey, ArgSlice destinationKey
 
             try
             {
-                var sremStatus = SetRemove(sourceKey, member, out var sremOps, ref objectLockableContext);
+                var arrDstKey = destinationKey.ToArray();
+                var arrSrcKey = sourceKey.ToArray();
 
-                if (sremStatus == GarnetStatus.NOTFOUND)
-                {
+                var srcGetStatus = GET(arrSrcKey, out var srcObject, ref objectLockableContext);
+
+                if (srcGetStatus == GarnetStatus.NOTFOUND)
                     return GarnetStatus.NOTFOUND;
-                }
 
-                if (sremOps != 1)
-                {
+                if (srcObject.garnetObject is not SetObject srcSetObject)
+                    return GarnetStatus.WRONGTYPE;
+
+                // If the keys are the same, no operation is performed.
+                var sameKey = sourceKey.ReadOnlySpan.SequenceEqual(destinationKey.ReadOnlySpan);
+                if (sameKey)
                     return GarnetStatus.OK;
+
+                var dstGetStatus = GET(arrDstKey, out var dstObject, ref objectLockableContext);
+
+                SetObject dstSetObject;
+                if (dstGetStatus == GarnetStatus.OK)
+                {
+                    if (dstObject.garnetObject is not SetObject tmpDstSetObject)
+                        return GarnetStatus.WRONGTYPE;
+
+                    dstSetObject = tmpDstSetObject;
                 }
+                else
+                {
+                    dstSetObject = new SetObject();
+                }
+
+                var arrMember = member.ToArray();
+
+                var removed = srcSetObject.Set.Remove(arrMember);
+                if (!removed) return GarnetStatus.OK;
```

Comment:
Do these *Move operations need a "bool removeSrcKey" if they go empty?

Reply:
Will resolve this an other similar scenarios in a separate PR